src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointJsonResponseEmitter.cs

```diff
@@ -5,27 +5,24 @@ namespace Microsoft.AspNetCore.Http.Generators.StaticRouteHandlerModel.Emitters;
 
 internal static class EndpointJsonResponseEmitter
 {
-    internal static void EmitJsonPreparation(this Endpoint endpoint, CodeWriter codeWriter)
+    internal static void EmitJsonPreparation(this EndpointResponse endpointResponse, CodeWriter codeWriter)
     {
-        if (endpoint.Response.IsSerializable)
+        if (endpointResponse is { IsSerializable: true, ResponseType: {} responseType })
```

Comment:
Is there a reason we are checking `ResponseType` for `null` here, but not below in `EmitJsonResponse`?

Reply:
We call `EmitJsonResponse` from the the main request handler where we do null checking on the type earlier on ([ref](https://github.com/dotnet/aspnetcore/blob/safia/rdg-nullability/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs#L86-L89)). 

We use null coalescing when we call `EmitJsonPreparation` [here](https://github.com/dotnet/aspnetcore/blob/safia/rdg-nullability/src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs#L88) so the null check isn't really required there.

Side note: I'm doing a lot of shuffling in this codepath in another branch as I'm working through a bug I found in writing JSON response for certain async handlers.